restaurant-menu/src/containers/Menu/Cart/Cart.js

```diff
@@ -0,0 +1,69 @@
+import './Cart.css';
+import React, {useState, useContext} from 'react';
+import OrderContext from '../../../OrderContext';
+import DishInCart from './DishInCart/DishInCart';
+import { useMediaQuery } from 'react-responsive'
+
+const Cart = props => {
+
+    const order = useContext(OrderContext)[0];
+    const delivery_costs = useContext(OrderContext)[2];
+    const isMobile = useMediaQuery({ query: '(max-width: 700px)' });
+
+    let initialWindowOpen = true;
+    if (isMobile)
+        initialWindowOpen = false;
+
+    const [windowOpen, setWindowOpen] = useState(initialWindowOpen);
+
+    let totalPrice = delivery_costs;
+
+    const addToPrice = dishInOrder => {
+        let addedPrice = parseFloat(dishInOrder.dish.price);
+        dishInOrder.extras.forEach(extra => {
+            addedPrice += parseFloat(extra.price);
+        });
+        totalPrice += addedPrice * dishInOrder.quantity;
+    };
+
+    const dishesInOrder = order.dishes_in_order.map(dishInOrder => {
+        addToPrice(dishInOrder);
+        return <DishInCart key={Math.random()} dishInCart={dishInOrder}/>
+    });
+
+    let dishesInOrderContent = (
+        <React.Fragment>
+            <div className="dishes-in-order">
+                {dishesInOrder} 
+            </div>
+            {totalPrice > delivery_costs ?
+            <React.Fragment>
+                <div className="total-price">
+                    <h4>{`סה"כ + משלוח ₪` + totalPrice}</h4>
+                </div>
+                <div className="submit-order" onClick={props.submitHandler}>בצע הזמנה</div>
+            </React.Fragment>
+            : <p>העגלה ריקה</p>
+            }
+        </React.Fragment>
+    );
+
+    if (!windowOpen && isMobile)
+        dishesInOrderContent = null;
+
+    return (
+        <aside className="cart">
+            <div className="title">
+                <h3>עגלת הזמנה</h3>
+                {windowOpen ?
+                    <i class="fas fa-minus" onClick={() => setWindowOpen(false)}></i>
+                :
+                    <i class="fas fa-plus" onClick={() => setWindowOpen(true)}></i>
+                }
+            </div>
+            {dishesInOrderContent}
+        </aside>
+    );
+}
+
+export default Cart;
```

Comment:
Unrelated to the above code:
![image](https://user-images.githubusercontent.com/20566827/125870551-19f20443-8de5-4892-a428-aadffc228271.png)
Why wouldn't this code work? What are some ways to fix it? One is using useRef, one is by changing the onEffect deps array, and one is by passing a function to setCounter.

Reply:
I didn't check it but I remember that:

First, we want the useEffect to run once, so we don't have multiple setIntervals. We can do that like you said with an empty deps array.
`useEffect( () => { ... }, []);`

When using setState with the value of the state, like you said, we should pass a function to setCounter.
`setCounter(prevCounter => prevCounter + 1)`

I'm not sure why we need useRef here.